Website: Add admin tool for generating Fleet Premium licenses.

eashaw opened the pull request:
Creating this as a draft PR for @mike-j-thomas's review. This PR will be ready for final review after his approval. 

Closes: https://github.com/fleetdm/confidential/issues/1744

Changes:
- Added `admin/generate-license` - an admin-only page for generating Fleet Premium licenses.
   - This page has a form with three inputs: organization name, number of hosts, and valid until date.
- Created a new action - `admin/generate-license-key.js` this action calls the create-license-key helper and returns the output.
- Updated policies and routes, and regenerated cloud-sdk


Note: To use this tool locally, `sails.config.custom.licenseKeyGeneratorPrivateKey` and `sails.config.custom.licenseKeyGeneratorPassphrase` need to be set. 
@mike-j-thomas Please let me know if you need help configuring these variables to test this.

mike-j-thomas commented:
> Note: To use this tool locally, sails.config.custom.licenseKeyGeneratorPrivateKey and sails.config.custom.licenseKeyGeneratorPassphrase need to be set.

@eashaw, please can you post instructions for how I can set up these variables? Thanks.

mike-j-thomas commented:
Hey @eashaw, sorry to keep you waiting on this review. I added the local config variables you provided, but I'm unsure how to access the page. I navigated to `admin/generate-license` but was redirected to `customers/login`. 

Let's go through it during our 1:1.

mike-j-thomas commented:
To avoid other users having the same confusion I had, let's do the following, as discussed in our 1:1.

TODO: Look into the possibility of adding a query string to `/customers/login?admin` that allows us to style the login screen conditionally.

Login screen design in Figma https://www.figma.com/file/yLP0vJ8Ms4GbCoofLwptwS/%E2%9C%85-fleetdm.com-(current%2C-dev-ready)?node-id=6692%3A24377

mike-j-thomas commented:
@eashaw, this is looking good, thanks.

**From Slack:**

> I was wrong about the redirects for the admin license generator. We can redirect logged-out users who go to the admin license generator to `/customers/login?admin` and show them a different version of the customer portal login.

This is great. Much better UX, thanks.

**Some feedback:**

1. What happens if **create account** is hit on `customers/login?admin`? Will that flow create an admin or a regular account? I assume that not just anyone can create an admin account, so should we consider conditionally hiding **create account**if that's the case?
2. When logged in, the browser back button directed me to `/sandbox-expired`.
3. Text change. Heading: `License key tool` Description: `Generate a Fleet Premium license key` Button: `Generate license key`. I updated Figma to reflect this change.
4. I suggest adding a text link to copy to the clipboard. It wasn't clear to me that I could click the text area. I found out by accident after highlighting all the text. Check Figma for button position.
5. From the generated license page, please add a button labeled **Generate another**, which resets the tool.

Updated Figma https://www.figma.com/file/yLP0vJ8Ms4GbCoofLwptwS/%E2%9C%85-fleetdm.com-(current%2C-dev-ready)?node-id=6692%3A24377

eashaw commented:
@mike-j-thomas I updated this PR with those requested changes. 

> 2. When logged in, the browser back button directed me to /sandbox-expired.

The view-action for the customer login page redirects logged-in users who have not purchased a license to Fleet sandbox, I've updated this to redirect admin users to the license generator.

hossein4667 reviewed (commented):
express
